rest-api-spec/src/main/resources/rest-api-spec/test/index/92_flat_object_support_doc_values.yml

```diff
@@ -0,0 +1,788 @@
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  prefix: {
+                    issue.labels: "abc8"
+                  }
+                }
+              ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+
+  # Regexp Query with exact dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    order: "order8"
+                  }
+                },
+                {
+                  regexp: {
+                    issue.labels.name: "ab.*"
+                  }
+                }
+              ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+  - match: { hits.hits.0._source.issue.labels.name: "abc8" }
+
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  regexp: {
+                    issue.labels.name: "a.*c8"
+                  }
+                }
+              ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+
+  # Regexp Query with no path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    order: "order8"
+                  }
+                },
+                {
+                  regexp: {
+                    issue.labels: "ab.*"
+                  }
+                }
+              ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  regexp: {
+                    issue.labels: "a.*c8"
+                  }
+                }
+              ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.order: "order8" }
+
+  # Fuzzy Query with exact dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    order: "order8"
+                  }
+                },
+                {
+                  fuzzy: {
+                    issue.labels.name: {
+                      value: "abcx",
+                      fuzziness: 1
+                    }
+                  }
```

Comment:
I'm kind of curious about this one. If you did a fuzzy query on `issue.labels.namx` for `abc8`, would it match? 

Unless I'm mistaken, it converts to a fuzzy query on `issue.labels._valueAndPath` for `namx=abc8`, which I think would match. Given that such a false positive is relatively unlikely, I'm wondering if we could use a `TwoPhaseIterator` to filter them out (without adding too much cost to the true positives).

I wouldn't treat it as a blocker for this PR, but we might want to document it as a known issue if we ship it with fuzzy matches on the dot path.

Reply:
@msfroh Yes, you are right, I tested that `abc8` can be match with the query `namx=abc8`.

That use `TwoPhaseIterator` to filter makes sense to me, If I fix it in the current pr, or create a new one?

Reply:
@msfroh please help see if there are any other problems?

Reply:
Yeah -- I would definitely fix it in a different PR (if we choose to fix it at all). 

As is, I believe fuzzy query is not used very often, so I think we could get away with documenting it as a known issue for now.